streams/src/main/java/org/apache/kafka/streams/processor/TimestampExtractor.java

```diff
@@ -17,12 +17,14 @@
 package org.apache.kafka.streams.processor;
 
 import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.annotation.InterfaceStability;
 import org.apache.kafka.streams.kstream.KTable;
 
 /**
  * An interface that allows the Kafka Streams framework to extract a timestamp from an instance of {@link ConsumerRecord}.
  * The extracted timestamp is defined as milliseconds.
  */
+@InterfaceStability.Evolving
```

Comment:
I am conservative on ts extractor since we may improve on timestamp management in the next release.

I'll add this to its impl classes as well.